core/templates/dev/head/components/ck-editor-helpers/ck-editor-5-rte.directive.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for CK Editor 5.
+ */
+
+require('services/ContextService.ts');
+require('services/RteHelperService.ts');
+
+const ClassicEditor = require(
+  '@ckeditor/ckeditor5-build-classic/build/ckeditor.js');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.directive('ckEditor5Rte', [
+  'ContextService', 'RteHelperService', 'PAGE_CONTEXT',
+  function(ContextService, RteHelperService, PAGE_CONTEXT) {
+    return {
+      restrict: 'E',
+      scope: {
+        uiConfig: '&'
+      },
+      template: '<div><div></div>' +
+                '<div contenteditable="true" class="oppia-rte">' +
+                '</div></div>',
+      require: '?ngModel',
```

Comment:
Why the "?" -- shouldn't all RTEs have a defined ngModel?

Reply:
yes, all RTEs need ngModel especially for plugins which have a modal for text input.
This is from CK-4 working because it was always able to find an ng-model controller.
let's make it `^^ngModel`? 
Also, do we have any other alternative to pass this?

Reply:
I'm not 100% sure what you should do, but try and do something that actually raises an error if the condition is not met, and that is as strict and narrowly-defined as possible. So, either "no prefix", or "^", or "^^".

Reply:
Hi @seanlip, I've used `^` learned for [here](https://docs.angularjs.org/guide/directive), this will throw an error is its not found thanks!